homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,218 @@
+            # what features the Innverter has.
+            if user_input[CONF_CAPABILITY] & ~CAPABILITY:
+                _errors[CONF_CAPABILITY] = "capability_invalid"
+            else:
+                self._data.update(user_input)
+                title = self._data[CONF_NAME]
+                return self.async_create_entry(title=title, data=self._data)
+        else:
+            # Use the capability bits from the Inverter. This assumes it
+            # had been possible to actually scrap them from the jScript
+            user_input = {}
+            user_input[CONF_MAX_OUTPUT] = self._enasolar.get_max_output()
+            user_input[CONF_DC_STRINGS] = self._enasolar.get_dc_strings()
+            user_input[CONF_CAPABILITY] = self._enasolar.get_capability()
+
+        return self.async_show_form(
+            step_id="inverter",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_MAX_OUTPUT, default=user_input[CONF_MAX_OUTPUT]
+                    ): vol.In(MAX_OUTPUT),
+                    vol.Required(
+                        CONF_DC_STRINGS, default=user_input[CONF_DC_STRINGS]
+                    ): vol.In(DC_STRINGS),
+                    vol.Required(
+                        CONF_CAPABILITY, default=user_input[CONF_CAPABILITY]
+                    ): vol.Coerce(int),
+                }
+            ),
+            errors=_errors,
+            last_step=True,
+        )
+
+
+class EnaSolarOptionsFlowHandler(config_entries.OptionsFlow):
+    """Allow Polling window to be updated."""
+
+    def __init__(self, config_entry):
+        """Initialize EnaSolar options flow."""
+        self.config_entry = config_entry
+        self._options = dict(config_entry.options)
+
+    async def async_step_init(self, user_input=None):
+        """Allow user to reset the Polling times."""
+
+        _errors = {}
+        if user_input is not None:
+            input_valid = True
+            if dt_util.parse_time(user_input[CONF_SUN_UP]) is None:
+                _errors[CONF_SUN_UP] = "time_invalid"
+                input_valid = False
+            if dt_util.parse_time(user_input[CONF_SUN_DOWN]) is None:
+                _errors[CONF_SUN_DOWN] = "time_invalid"
+                input_valid = False
+            if input_valid:
+                if dt_util.parse_time(user_input[CONF_SUN_UP]) >= dt_util.parse_time(
+                    user_input[CONF_SUN_DOWN]
+                ):
+                    _errors[CONF_SUN_DOWN] = "time_range"
+                    input_valid = False
+            if input_valid:
+                self._options.update(user_input)
+                return self.async_create_entry(title="", data=self._options)
+        else:
+            user_input = {}
+            if not self._options:
+                user_input[CONF_SUN_UP] = DEFAULT_SUN_UP
+                user_input[CONF_SUN_DOWN] = DEFAULT_SUN_DOWN
```

Comment:
Suggest to handle it in the schema instead with user_input.get(CONF_SUN_UP, DEFAULT_SUN_UP)

Reply:
Rewritten but still not sure am doing it right

Reply:
you can remove this else completely and change as per below